tutorials/ios-chatbot/index.md

```diff
@@ -0,0 +1,164 @@
+
+The next step after that is to add some more text responses. 
+
+![Default Welcome Intent Screenshot](chatbots-6.png "Default Welcome Intent Screenshot")
+
+
+Next, it's time to work on contexts.
+
+
+### Contexts
+
+_[Contexts](https://docs.api.ai/docs/concept-contexts)_ represent the current context of a user's request. They are helpful for differentiating phrases that may be vague or have different meanings depending on the user's preferences or geographic location, the current page in an app, or the topic of conversation. Let's look at an example.
+
+_User: Where am I?_
+
+_Bot: Please upload a nearby picture and I can help find out where you are._
+
+_[User uploads a picture of Golden Gate Bridge.]_
+
+_Bot: You are near Golden Gate Bridge._
+
+_User: How much is the ticket?_
+
+_Bot: Golden Gate Bridge is free to visit._
+
+_User: When does it close today?_
+
+_Bot: Golden Gate Bridge is open 24 hours a day, 7 days a week._
+
+_User: How do I get there?_
+
+_[Bot shows a map to Golden Gate Bridge.]_
+
+In the above conversation, when the user asks "How much is the ticket?" and "When does it close today?" or "How do I get there?", the bot understands that the context is around Golden Gate Bridge.
+
+The next thing to do is to weave intents and contexts together. For our example, each box in the diagram below is an intent and a context; the arrows indicate the relationships between them.
+
+
+
+![alt_text](chatbots-7.png "Contexts Relationship")
+
+
+
+### Output Contexts
+
+Contexts are tied to user sessions (a session ID that you pass in API calls). If a user expression is matched to an intent, the intent can then set an output context to be shared by this expression in the future. You can also add a context when you send the user request to your API.AI agent. In our example, the `where` intent sets the `where` output context so that _location_ intent will be matched in the future.
+
+
+### Input Contexts
+
+Input contexts limit intents to be matched only when certain contexts are set. In our example, location's input context is set to where. The `location` intent is matched only when we are under `where` context.
+
+Here are the steps to generate these intents and contexts:
+
+
+
+1.  create `where` intent and add `where` output context. This is the root in the context tree and has no input context.
+
+![alt_text](chatbots-2.png "Contexts Screenshot")
+
+
+1.  create `location` intent. 
```

Comment:
The numbering here is still off. Perhaps if you removed the extra line break between the steps.

Reply:
@PokerChang